### ISSUE TITLE
CuDNN RNN layers nested in Model are not converted when loading

### ISSUE THREAD

**Issue by bzamecnik** [keras-team/keras#10080]
Analogous to CuDNN RNN nested in Bidirectional: #8908

In addition same bug as in: Convert h5py Dataset to np.array before preprocess_weights_for_loading (#9662)

I've already wrote a fix and I'll post a detailed description soon.

```
ValueError: Shapes must be equal rank, but are 2 and 1 for 'Assign_732' (op: 'Assign') with input shapes: [2,384], [768].
```

Bias is not converted from `CuDNNGRU` shape (2*units*gates) to CuDNN-compatible, `GRU(reset_after=True` shape (2, units*gates). The reason that nesting to Model is missing in `keras.topology.saving.preprocess_weights_for_loading()`.

We can do the same as in the block for converting from Keras 1 model.

In addition `weights[2]` is H5 dataset:

```
biases = weights[2].reshape((2, -1) if from_cudnn else -1)
```

and fails with:

```
'Dataset' object has no attribute 'reshape'
```

Fix is:

```
biases = np.array(weights[2]).reshape((2, -1) if from_cudnn else -1)
```

**Comment by Ajk4** [comment]
Hi @bzamecnik . Thank you for your PR!

FIY I installed keras from sources with all those pull requests and it seems like the following corner case is unhandled:

  **Nested model with gru rnn used within TimeDistributed**

Code example:

```
# encoder_model has GRU or CUDNNGRU depending on flag
encoded_input = TimeDistributed(encoder_model)(input)
```

When I try to load CUDNNGRU weights to GRU(reset_after=True) I get:

```
ValueError: Layer #1 (named "time_distributed_6"), weight <tf.Variable 'encoder_rnn_01/bias:0' shape=(2, 1536) dtype=float32_ref> has shape (2, 1536), but the saved weight has shape (3072,).
```

I guess we need additional else statement in https://github.com/keras-team/keras/blob/ce56322aadc6a861db21e598720c2dce03747ec3/keras/engine/saving.py#L541
for TimeDistributed too.

**Comment by bzamecnik** [comment]
@Ajk4 Aah, yes. `TimeDistributed` is another wrapped that needs to be traversed in the weight conversion process (besides `Bidirectional` and `Model`). Are there any other possibilities left so that we can fix them at once?

**Comment by Ajk4** [comment]
I'm not sure since I'm not that familiar with keras source code. However I feel like its the last missing piece here.

**Comment by bzamecnik** [comment]
Closing this issue since it was fixed in PR #10081.
